### DISCUSSION TITLE
How to use regression in Avalanche

### DISCUSSION THREAD

**Discussion by raul-parada** [ContinualAI/avalanche#1465] [Q&A]
I would like to adapt Avalanche in my code. The dataset is composed of Cartesian coordinates in which X and y are the same. I've solved the fact of using a 2D in the y with this simple solution.

    X = raw[:-1]
    y = raw[1:]

Mention these two lines above allow that both latitude and longitude can be used as data and label. Since coordinates are a unique pair of lat/lon, I cannot apply a label to each cartesian coordinates

This is the code:

    #!pip install haversine
    from sklearn.linear_model import SGDRegressor
    from sklearn.linear_model import LinearRegression
    from sklearn.metrics import mean_squared_error
    import haversine as hs
    from haversine import Unit
    num=10
    
    i=1
    
    while((i*num+1)<len(10)):
      raw= dat[0:(i*num)]
      X = raw[:-1]
      y = raw[1:]
      regressor = SGDRegressor(max_iter=1000, tol=1e-3, random_state=42)
      #regressor = LinearRegression()
      regressor.partial_fit(X, y)
    
      pred=regressor.predict([raw[(i*num)-1]])
      hs.haversine(( pred[0,0] , pred[0,1] ),( dat[(i*num),0] , dat[(i*num),1] ), unit=Unit.METERS)
      i=i+1

The piece of data can be obtained from here:

    import pandas as pd
    import io
    
    content = """latitude(m) longitude(m)
    41.392245 2.162762
    41.392261 2.162783
    41.392292 2.162825
    41.392340 2.162888
    41.392369 2.162943
    41.392405 2.163008
    41.392454 2.163093
    41.392520 2.163186
    41.392600 2.163296
    41.392687 2.163417"""
    
    # Create a file-like object from the content string
    file_obj = io.StringIO(content)
    
    # Load the content into a DataFrame
    raw = pd.read_csv(file_obj, delimiter='\t' or ' ')

I've already installed Avalanche and imported the useful libraries. However, I don't know how to do regression using this simple code into Avalanche. Please, could help me on this? Thank you.
